packages/codegen.go/src/operations.ts

```diff
@@ -738,42 +738,95 @@ function createProtocolRequest(azureARM: boolean, method: go.MethodType | go.Nex
     hostParam = `runtime.JoinPaths(${hostParam}, urlPath)`;
   }
 
+  // helper to build nil checks for param groups
+  const emitParamGroupCheck = function (param: go.MethodParameter): string {
+    if (!param.group) {
+      throw new CodegenError('InternalError', `emitParamGroupCheck called for ungrouped parameter ${param.name}`);
+    }
+    let client = '';
+    if (param.location === 'client') {
+      client = 'client.';
+    }
+    const paramGroupName = uncapitalize(param.group.name);
+    let optionalParamGroupCheck = `${client}${paramGroupName} != nil && `;
+    if (param.group.required) {
+      optionalParamGroupCheck = '';
+    }
```

Comment:
Nit: maybe a meaningless perf optimization. Could set to '' and the only do the larger replacement if !pararm.group.required